YouVsVirus/Assets/Scenes/LevelCollectMasks/CreatePopLevelcollectmasks.cs

```diff
@@ -0,0 +1,151 @@
+            LevelSettings.GetActiveLevelSettings().NumberOfNPCs = npcNumber;
+            // We do not show the infection status in this level
+            LevelSettings.GetActiveLevelSettings().ShowInfectionStatus = false;
+            // Set the DayLength for this level
+            LevelSettings.GetActiveLevelSettings().DayLength = 100;
+            // this gets the Main Camera from the Scene
+            // the grid cell has to be as large as the player's infection radius
+            randomGridForHumans = GameObject.Find("RandomGrid").GetComponent<RandomGrid>();
+            // make screen Bounds 80% smaller so that NPCs are placed more in the middle since the player is at the edge
+            randomGridForHumans.shrinkScreenBounds(0.8f);
+
+            // generate the random coordinates for humans which depend on the scale of the player (who is largest),
+            // their infection radius, since we do not want immediate infection 
+            // and the number of humans that we want to place
+            randomGridForHumans.GenerateRandomCoords(playerPrefab.transform.localScale.x,
+                                                            playerPrefab.GetComponentInChildren<InfectionTrigger>().InfectionRadius,
+                                                            npcNumber, nonSpawnableSpaceClass);
+            // place humans on grid
+            CreateHumans();
+        }
+
+        /// <summary>
+        /// Places the player and all NPCs on the map randomly.
+        /// </summary>
+        private void CreateHumans()
+        {
+            Vector3 coords3D = new Vector3 (PlayerSpawnCoordinates[0], PlayerSpawnCoordinates[1], 0);
+            //  Place the player
+            Player = Instantiate(playerPrefab.GetComponent<Player>(),
+                                 coords3D,
+                                 Quaternion.identity);
+            // give the player a unique id
+            Player.myID = 0;
+            // Set player status to inactive. Will get avtivated when we leave the house
+            Player.gameObject.SetActive(false);
+
+            //  Place the NPCs in the grid
+            for (int i = 1; i <= npcNumber; i++)
+            {
+                NPCs.Add(Instantiate(npcPrefab.GetComponent<NPC>(),
+                                        randomGridForHumans.RandomCoords[i],
+                                        Quaternion.identity));
+                // give all npcs a unique id
+                NPCs[i - 1].myID = i + 1;
+            }
+
+            //  Infect one
+            NPCs[Mathf.RoundToInt(npcNumber/2f)].SetInitialCondition(NPC.EXPOSED);
+        }
+
+        void Update()
+        {
+            if (Input.GetKeyDown(KeyCode.Space))
+            {
+                // player is no longer shown in the house
+                PlayerHouse.GetComponent<PlayerHouse>().UnshowPlayer();
+                // can start to play
+                Player.gameObject.SetActive(true);
```

Comment:
Did you not tell me that it is better to use .enabled ?

Reply:
That was for the sprite renderer, where the alternative was setting it to a different layer.

I guess SetActive is the proper way to go here, since it activates the whole game object.
This way, the player just does not exist when he is inside the house.
Which is good, since otherwise it could still move around, even though invisible.